client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx

```diff
@@ -175,19 +197,124 @@ const getEmailForwardMenuItems = ( { dispatch, mailbox, translate } ) => {
 	];
 };
 
+const RemoveTitanMailboxConfirmationDialog = ( { mailbox, visible, setVisible } ) => {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	const emailAddress = getEmailAddress( mailbox );
+
+	const noticeDuration = 7000;
+
+	const errorMessage = errorNotice(
+		translate(
+			'There was an error removing {{strong}}%(emailAddress)s{{/strong}} from your account',
+			{
+				comment: '%(emailAddress)s is the email address associated to the mailbox being deleted',
+				args: { emailAddress },
+				components: {
+					strong: <strong />,
+				},
+			}
+		),
+		{ duration: noticeDuration }
+	);
+
+	// The mailboxes are not removed immediately, but rather scheduled. An async job takes care of the deletion. Then
+	// we also wait for the deletion event to come through from Titan since we really only read the local tables.
+	const successMessage = successNotice(
+		translate(
+			'{{strong}}%(emailAddress)s{{/strong}} has been scheduled for removal from your account',
+			{
+				comment: '%(emailAddress)s is the email address associated to the mailbox being deleted',
+				args: { emailAddress },
+				components: {
+					strong: <strong />,
+				},
+			}
+		),
+		{ duration: noticeDuration }
+	);
+
+	const { removeTitanMailbox } = useRemoveTitanMailboxMutation( mailbox.domain, mailbox.mailbox, {
+		onSettled: ( data ) => {
+			if ( data?.status === 202 ) {
+				dispatch( successMessage );
+				return;
+			}
+			dispatch( errorMessage );
+		},
+	} );
+
+	const onClose = ( action ) => {
+		setVisible( false );
+
+		if ( 'remove' === action ) {
+			removeTitanMailbox();
+
+			recordTracksEvent( 'calypso_email_management_titan_remove_mailbox_execute', {
+				domain_name: mailbox.domain,
+				mailbox: mailbox.mailbox,
+			} );
+		}
+	};
+
+	const buttons = [
+		{ action: 'cancel', label: translate( 'Cancel' ) },
+		{ action: 'remove', label: translate( 'Confirm' ), isPrimary: true },
+	];
+
+	return (
+		<Dialog
+			className="email-mailbox-action-menu__remove-titan-mailbox-dialog"
+			isVisible={ visible }
+			buttons={ buttons }
+			onClose={ onClose }
+		>
+			<div>
+				<h3> { translate( 'Remove mailbox' ) } </h3>
+				<p>
+					{ translate(
+						'Are you sure you want to remove {{strong}}%(emailAddress)s{{/strong}}? All your data will be deleted!',
```

Comment:
Should we be specific about what "data" is involved? Might it be clearer to say something like:
> All your emails, calendar events, and contacts will be deleted!

This is non-blocking.